capstone-frontend/src/components/Browse.jsx

```diff
@@ -1,10 +1,103 @@
-import React from 'react';
-import BookSearchList from './BookSearchList';
+import React, { Component } from 'react';
+import { Nav, Row, Spinner, Tab } from 'react-bootstrap';
+import { BookSearchList } from './BookSearchList';
+import { ClubSearchList } from './ClubSearchList';
 
-const Browse = (props) => {
-  return (
-    <BookSearchList searchQuery={props.searchQuery} bookLists={props.bookLists} updateBookLists={props.updateBookLists} />
-  );
-}
+import '../styles/Browse.css';
 
-export default Browse;
\ No newline at end of file
+export class Browse extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchedBooks: [],
+      loading: true, // Spinner
+    }
+  }
+
+  getSearchedBooks = (searchQuery) => {
+    fetch(`/api/search?searchTerm=${searchQuery}`)
+      .then(response => response.json())
+      .then(searchedBooks => this._isMounted && this.setState({ searchedBooks, loading: false }))
+      .catch(function (err) {
+        alert(err)
+      });
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+
+    if (this._isMounted) {
+      this.setState({ loading: true });
+      this.getSearchedBooks(this.props.searchQuery);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.searchQuery !== prevProps.searchQuery) {
+      this.setState({ loading: true });
+      this.getSearchedBooks(this.props.searchQuery);
+    }
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  render() {
+    return (
+      <>
```

Comment:
I don't think the fragment is necessary here if everything is wrapped in the `Tab.Container`

Reply:
Right! This makes sense, thanks for pointing it out